ironfish/src/chainProcessor.ts

```diff
@@ -92,6 +101,11 @@ export class ChainProcessor {
       await this.remove(remove)
       this.hash = remove.previousBlockHash
       this.sequence = remove.sequence - 1
+
+      blockCount++
+      if (this.maxQueueSize && blockCount >= this.maxQueueSize) {
+        return { hashChanged: !oldHash || !this.hash.equals(oldHash) }
```

Comment:
Nit: given that this is repeated in multiple places, we could have an anonymous fn like this:

```ts
const returnValue = () => ({ hashChanged: !oldHash || !this.hash.equals(oldHash) })
```

and then change all the `return { ... }` statements to:

```ts
return returnValue()
```

Reply:
I have one more recommendation for this that I put in another PR here, https://github.com/iron-fish/ironfish/pull/5073.